### DISCUSSION TITLE
Loading a pattern file based on file extension?

### DISCUSSION THREAD

**Discussion by graydav1** [WerWolv/ImHex#1397] [Pattern Language Q&A] [answered]
As the title suggests, is there a way to get a pattern to automatically load based on the extension of the file?

I'm working with some raw serial port bytes I've dumped from an app I'm writing, and the file has no mime type, but will always have the same extension, so it'd be nice to be able to load the data with something like a `#pragma extension .myfiledump` or similar. 

Can this be done?

**Comment by WerWolv** [comment, marked as answer]
Hey
This is generally not supported as file extensions don't really mean anything and not all providers even have the concept of file extensions.
Instead you can either write a custom magic file and assign a MIME type to your file that way or you use the new `#pragma magic [ AA ?? ?? DD ] @ 0x12345678` pragma.

**Reply by graydav1** [reply]
That worked perfectly. Thanks!

And thanks for the quick replies. I really appreciate the effort :)
